### PR TITLE
Resource types screen

### PR DESCRIPTION
**Pull request by arturdryomov**
I tried `RecyclerView` and it remains totally gross regarding simple things — there is no simple ability to add list dividers or just set an item click listener.

Not sure if `BackendClient` should be a singleton really — AeroGear components handle most of the state: `PipeManager`, `AuthorizationManager`. All these `ReadFilter`-related things are potentially not very nice — I had a plan to declare all paths at one place but probably it is not possible. I’ll think about it.

### CODE REVIEW

**Comment by danielpassos** [comment]
> I tried `RecyclerView` and it remains totally gross regarding simple things — there is no simple ability to add list dividers or just set an item click listener.

Yes, there is. We can create a [RecyclerView.OnItemTouchListener](http://developer.android.com/reference/android/support/v7/widget/RecyclerView.OnItemTouchListener.html) to do the same what [ListView.OnItemClickListner](https://developer.android.com/reference/android/widget/AdapterView.OnItemClickListener.html) do.

See my gist: https://gist.github.com/danielpassos/6aec0da5f53679ae23b6

**Comment by danielpassos** [comment]
> Not sure if `BackendClient` should be a singleton really — AeroGear components handle most of the state: `PipeManager`, `AuthorizationManager`. 

Yeap, you don't need to worry about your Pipe, Auth, etc, created.

**Comment by danielpassos** [comment]
> All these `ReadFilter`-related things are potentially not very nice — I had a plan to declare all paths at one place but probably it is not possible. I’ll think about it.

I'm not following you here. What it means? What do you need in `ReadFilter`?

**Comment by danielpassos** [comment]
I think you can create/store all Pipes, Auths you need in `Application.onCreate` instead of use `BackendClient`, but let's do it step by step.

**Comment by arturdryomov** [comment]
> Yes, there is. We can create a `RecyclerView.OnItemTouchListener` to do the same what `ListView.OnItemClickListener` do.

Thanks, but I know it is possible, it is just not justified to abandon `ListView`, especially when no benefits of `RecyclerView` will be used (probably).

> I'm not following you here. What it means? What do you need in `ReadFilter`?

Well, I thought about putting all paths into the `BackendPipes.Paths` class. But Hawkular paths require variable components, i. e. `/inventory/{tenantId}/resourceTypes`. It is not a great idea to create a new pipe for every one of them, i. e. `/inventory/xxxx-xxxx/resourceTypes` and `/inventory/yyyy-yyyy/resourceTypes`. `ReadFilter` comes in play, but it blocks my idea about all paths in one place. Maybe I should workaround using another way.
- Register pipes with a common path via `PipeManager.config`. For example, tenants, resources and metrics are placed under `https://server.link/hawkular/inventory`, so it will be used with `PipeManager.config` for tenants, resources and metrics as well.
- Fetch pipes using `ReadFilter`. For example, tenants — `tenant/`, resource types — `{tenantId}/resourceTypes`, metrics — `{tenantId}/resourceTypes/{resourceType}`.

What do you think?

> I think you can create/store all Pipes, Auths you need in `Application.onCreate` instead of use `BackendClient,` but let's do it step by step.

Not exactly — the endpoint should be able to change at runtime, so some sort of re-registration should be available.

**Comment by arturdryomov** [comment]
Ping @danielpassos.

File: src/main/java/org/hawkular/client/android/util/Intents.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+			return new Builder(context);
+		}
+
+		private Builder(Context context) {
+			this.context = context;
+		}
+
+		public Intent buildResourceTypesIntent(@NonNull Tenant tenant) {
+			Intent intent = new Intent(context, ResourceTypesActivity.class);
+			intent.putExtra(Extras.TENANT, tenant);
```
Comment (danielpassos) [inline src/main/java/org/hawkular/client/android/util/Intents.java:37]: The builder is only to create a intent for tenant?

Comment (arturdryomov) [inline src/main/java/org/hawkular/client/android/util/Intents.java:37]: I have a plan to have all intents there and names for extras as well.

**Pull request merged**